File: backend/app/schemas/public.py
from datetime import date
from uuid import UUID

from pydantic import BaseModel


class PublicDocument(BaseModel):
    id: UUID
    document_type: str
    title: str
    activity_date: date | None
    issue_date: date
    status: str


class StudentDocumentsResponse(BaseModel):
    full_name: str
    roll_number: str
    activity_certificates: list[PublicDocument]
    leadership_recognition: list[PublicDocument]

File: frontend/src/api/documents/public.ts
export type PublicDocument = { id: string; title: string; issue_date: string; status: string };
export type StudentDocuments = { full_name: string; roll_number: string; activity_certificates: PublicDocument[]; leadership_recognition: PublicDocument[] };
const apiBase = import.meta.env.VITE_API_BASE_URL ?? "/api";
export async function getStudentDocuments(rollNumber: string): Promise<StudentDocuments> {
  const response = await fetch(`${apiBase}/public/students/${encodeURIComponent(rollNumber)}/documents`);
  if (!response.ok) throw new Error(response.status === 404 ? "No certificates or letters are currently available for this roll number." : "Unable to load records right now.");
  return response.json();
}
